sprint0/sprint0/Bosses/Dodongo/Dodongo.cs

```diff
@@ -58,100 +60,185 @@ public Dodongo(Texture2D texture, Vector2 location)
+                    if (Location.X >= (Game1.Width - 50) * Game1.Scale)
+                    {
+                        FaceNorth();
+                    }
+                }
+                else if (direction == Direction.s)
                 {
-                    direction = Direction.n;
-                    currentFrameUD = (totalFramesUD * repeatedFrames / 2) + 1;
-                    Location = new Rectangle(Location.X, Location.Y, sideLength, Location.Height); // change to vertical dimensions
+                    //animates sprite by fliping after every repeatedFrames frames
+                    currentSpriteEffect = (currentSpriteEffect + 1) % (totalSpriteEffects * repeatedFrames);
+
+                    //moves sprite down
+                    Location = new Rectangle(Location.X, Location.Y + 1, Location.Width, Location.Height);
+                    if (Location.Y >= (Game1.HUDHeight + Game1.MapHeight - 50) * Game1.Scale)
+                    {
+                        FaceEast();
+                    }
+                }
+                else
+                {   //direction == Direction.n
+
+                    //animates sprite by flipping after every repeatedFrames frames
+                    currentSpriteEffect = (currentSpriteEffect + 1) % (totalSpriteEffects * repeatedFrames);
+
+                    //moves sprite up
+                    Location = new Rectangle(Location.X, Location.Y - 1, Location.Width, Location.Height);
+                    if (Location.Y <= (Game1.HUDHeight + 50) * Game1.Scale)
+                    {
+                        FaceWest();
+                    }
                 }
             }
-            else if (direction == Direction.s)
+            else // eating; do not move
             {
-                //animates sprite by fliping after every repeatedFrames frames
-                currentSpriteEffect = (currentSpriteEffect + 1) % (totalSpriteEffects * repeatedFrames);
-
-                //moves sprite down
-                Location = new Rectangle(Location.X, Location.Y + 1, Location.Width, Location.Height);
-                if (Location.Y >= (Game1.HUDHeight + Game1.MapHeight - 50) * Game1.Scale)
+                eatingCounter = (eatingCounter + 1) % eatingTime;
+                if (eatingCounter == 0) // done eating; return to normal frames
                 {
-                    direction = Direction.e;
-                    currentFrameRL = 0;
-                    Location = new Rectangle(Location.X, Location.Y, width, Location.Height); // change to horizontal dimensions
+                    FaceDirection(direction);
                 }
-
             }
-            else
-            {   //direction == Direction.n
+        }
 
-                //animates sprite by fliping after every repeatedFrames frames
-                currentSpriteEffect = (currentSpriteEffect + 1) % (totalSpriteEffects * repeatedFrames);
+        private void FaceDirection(Direction dir)
```

Comment:
This class is very long and I feel like cohesion is pretty low. I know its pretty complicated but maybe in the future we can refactor this better. Maybe take change direction out into another class since lots of things use it. Idk. But good job getting it to work!